### ISSUE TITLE
As a user, I would like to search for instructors by their first and/or full name, instead of only last name.

### ISSUE THREAD

**Issue by andrewhlu** [ucsb-cs56-w20/ucsb-courses-search#127]

**Comment by ishana-narayanan** [comment]
1. Split the instructor string so that there is one for first name and another for last name (InsSearch.java)
2. Configure the drop down menu and autocomplete to include both first and last names of instructors

**Comment by pconrad** [comment]
@andrewhlu I got a question from @BrandonBai18 about this issue.. he's working on it.

The API only allows one search field, and that search field only has the names that you see on GOLD and on this site: <https://my.sa.ucsb.edu/public/curriculum/coursesearch.aspx>

We literally don't have any other data.  Nor it is clear where we would get data to populate a drop down, unless we used the pre-scraping approach that we piloted last week (i.e. the MongoDB database).

This team may need some more guidance on what you have in mind for this issue and how to proceed.

**Comment by pconrad** [comment]
It is NOT THE CASE that you can assume that the first space separates the last name from the first name.

As an example, see the course "MATH CS 108A" in W20.

The field has "BUENO CACHADI".

But it is NOT the case that the instructors last name is BUENO and the first name is CACHADI.

Instead, the last name is "BUENO CACHADINA" and the first names are "MARIA ISABELLA".

So, I think the entire premise of this issue may be faulty.

**Comment by pconrad** [comment]
I wonder if autocomplete (which @BrandonBai18 says is already done) might be sufficient to address the user need.. and maybe this isn't really adding much value (plus we don't know how to do it lol)

**Comment by andrewhlu** [comment]
The original idea behind this story was to create some internal database of instructors that mapped between an instructor's full name (e.g. "Phillip Conrad") and an instructor's name on GOLD (e.g. "CONRAD P T"). That way, we could perform searches using the instructor full names (and from that, their first name and last name), and then we would map that internally to performing an API search with the GOLD version of their name. The GOLD names would come from the scraping approach.

However, I took a look at the data returned by the API again, and you are correct about the fact that the API only returns names that we see on GOLD.

If we were to go forward with this, we would have to find a way to get an instructor's full name from their GOLD name. This isn't possible with only the data from the UCSB Developer API, so we would have to introduce some other service that could get those for us. 

One source that I found the other day was to use the [UCSB People Finder](http://www.identity.ucsb.edu/people_finder/) for UCSB faculty and staff to pull an instructor's full name, email, and more, using their GOLD name. (e.g. [this query](http://www.identity.ucsb.edu/people_finder/?rs=search_person_records&rsargs=%5B%22person%22%2C%22conrad%20p%22%5D)). It looks like JSON, but we'll have to do some formatting to get a usable result that we can convert.
